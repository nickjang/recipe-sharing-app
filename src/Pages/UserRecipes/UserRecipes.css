h2.author-name {
  color: var(--color-white);
  text-align: center;
}

@media all and (min-width: 1280px) {
  h2.author-name {
    text-align: left;
    margin-left: 7rem;
  }
}
