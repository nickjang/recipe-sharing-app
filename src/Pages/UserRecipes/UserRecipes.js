import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import RecipeSharingApiService from '../../services/recipe-sharing-api-service';
import InfiniteScroll from 'react-infinite-scroll-component';
import './Recipes.css';

class UserRecipes extends Component {
  state = {
    recipes: [],
    hasMore: false,
    author: '',
    error: ''
  }

  async fetchRecipes() {
    this.setState({ error: '' });
    try {
      const userId = this.props.match.params.userId;
      let { recipes, hasMore } = await RecipeSharingApiService.getMoreRecipes(12, userId);
      recipes = this.state.recipes.concat(recipes)
      this.setState({ recipes, hasMore });
    } catch ({ error }) {
      this.setState({ error, hasMore: false });
    }
  }

  async componentDidMount() {
    this.setState({ error: '' });
    try {
      const userId = this.props.match.params.userId;
      const { full_name } = await RecipeSharingApiService.getAuthor(userId);
      this.setState({ author: full_name });
    } catch ({ error }) {
      this.setState({ error });
    }
  }

  render() {
    const { recipes, hasMore, author, error } = this.state;

    return (
      <>
        <div>
          <h2>{author}</h2>
        </div>
        <InfiniteScroll
          dataLength={recipes.length}
          next={this.fetchRecipes}
          hasMore={hasMore}
          loader={<span>Loading...</span>}
          endMessage={error
            ? <p className='error'>{error}</p>
            : <p>End of recipes</p>}
        > {recipes.map((recipe) =>
          // style to show hand/link
          <article onClick={this.props.history.push(`/recipes/${recipe.id}`)}>
            <p>{recipe.name}</p>
            <p>{`Author: ${recipe.author.full_name}`}</p>
            <p>{recipe.information}</p>
            <ul>
              {recipe.ingredients.map(ingredient =>
                <li>{ingredient}</li>
              )}
            </ul>
          </article>
        )}
        </InfiniteScroll >
      </>
    );
  }
}

export default withRouter(UserRecipes);