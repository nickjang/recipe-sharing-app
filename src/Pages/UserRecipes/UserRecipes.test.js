import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserRecipes from './UserRecipes';

describe(`UserRecipes component`, () => {
  it('renders UserRecipes by default', () => {
    const wrapper = shallow(<UserRecipes />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})