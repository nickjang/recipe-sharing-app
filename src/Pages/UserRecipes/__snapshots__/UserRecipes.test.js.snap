// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRecipes component renders UserRecipes by default 1`] = `
<Fragment>
  <Parallax
    bgClassName="react-parallax-bgimage"
    bgImage="https://preppykitchen.com/wp-content/uploads/2019/06/Ultimate-Chocolate-Cake-feature-1200.jpg"
    bgImageAlt="the cat"
    blur={0}
    className=""
    contentClassName=""
    disabled={false}
    strength={200}
  >
    <h2
      className="rs-title author-name"
    />
  </Parallax>
  <InfiniteScroll
    className="recipe-card-list"
    dataLength={0}
    endMessage={
      <p
        className="note text-center"
      >
        End of recipes
      </p>
    }
    hasMore={true}
    loader={
      <span>
        Loading...
      </span>
    }
    next={[Function]}
  >
     
  </InfiniteScroll>
</Fragment>
`;
